go/arrow/type_traits_decimal128.go

```diff
@@ -39,8 +40,13 @@ func (decimal128Traits) BytesRequired(n int) int { return Decimal128SizeBytes *
 
 // PutValue
 func (decimal128Traits) PutValue(b []byte, v decimal128.Num) {
-	binary.LittleEndian.PutUint64(b[:8], uint64(v.LowBits()))
-	binary.LittleEndian.PutUint64(b[8:], uint64(v.HighBits()))
+	if runtime.GOARCH == "s390x" {
```

Comment:
Is this a compile time check?  Will it cause a branch at runtime?

Reply:
Good point.

Can we use `const` at the package level? For example,
```
package ...

const BigEndian = runtime.GOARCH == "s390x"

```

Reply:
`runtime.GOARCH` is "recorded at compile time and made available by constants or functions in this package, but they do not influence the execution of the run-time system" - according to https://golang.org/pkg/runtime/

Reply:
I found this an issue discussing [DCE](https://github.com/golang/go/issues/21424) in Go.  Skimming the thread it seems that using build tags if feasible might be a better way to support multiple architectures.

Reply:
A simple 'endian' (or another name) package containing some files like this might help reduce the if statement spam, and address @emkornfield's concerns (though the compiler is pretty good at removing those branches):

big.go:
```go
// +build s390x

package endian
import "encoding/binary"
var Native = binary.BigEndian
```
little.go:
```go
// +build !s390x

package endian
import "encoding/binary"
var Native = binary.LittleEndian
```

Then you can just use `endian.Native.*` functions instead of having if statements. Plus support for other big endian platforms becomes easy.

(Note that even though Native is a variable the methods called on it will still be resolved at compile time because the type is constant).

Reply:
This package would also be a good place for a `IsBigEndian` constant.

Reply:
Thanks @mundaym I'll update the commit